### PR TITLE
Raptor: bug on main_stop_areas cost

### PR DESCRIPTION
**Pull request by pbougue**
Following http://jira.canaltp.fr/browse/NAVP-226
Some stop_areas had a cost while being main_stop_areas of given departure admin
To merge after (update artemis test) : CanalTP/artemis#179

### CODE REVIEW

File: source/georef/tests/georef_test.cpp [hunk 1]
```diff
@@ -398,19 +398,21 @@ BOOST_AUTO_TEST_CASE(compute_nearest){
 
     w.init(starting_point);//not mandatory, but reinit to clean the distance table to get fresh dijsktra
     res = w.find_nearest_stop_points(100_s, pl, false);
-    BOOST_REQUIRE_EQUAL(res.size(), 1);
-    BOOST_CHECK_EQUAL(res[0].first , 0);
-    BOOST_CHECK_EQUAL(res[0].second, navitia::seconds(60 / (default_speed[Mode_e::Walking] * 2))); //the projection is done with the same mean of transport, at the same speed
+    //the projection is done with the same mean of transport, at the same speed
+    navitia::routing::map_stop_point_duration tested_map;
+    tested_map.insert(navitia::routing::stop_point_duration(navitia::routing::SpIdx(0),
```
Comment (antoine-de) [inline source/georef/tests/georef_test.cpp]: I think it's clearer if you build the SpIdx directly with the stop point

`navitia::routing::SpIdx(sp1)`

File: source/routing/raptor_utils.h [hunk 2]
```diff
@@ -48,6 +49,9 @@ typedef Idx<type::JourneyPatternPoint> JppIdx;
 typedef Idx<type::JourneyPattern> JpIdx;
 typedef Idx<type::StopPoint> SpIdx;
 
+typedef boost::container::flat_map<SpIdx, navitia::time_duration> map_stop_point_duration;
+typedef map_stop_point_duration::value_type stop_point_duration;
```
Comment (xlqian) [inline source/routing/raptor_utils.h]: FYI: http://en.cppreference.com/w/cpp/language/type_alias 
PS: there are absolutely no differences between both.

File: source/routing/tests/routing_api_test.cpp [hunk 3]
```diff
@@ -799,10 +799,11 @@ BOOST_FIXTURE_TEST_CASE(direct_path_under_the_limit, direct_path_routing_api_dat
...
 
-    BOOST_REQUIRE_EQUAL(sp.size(), 1);
-    auto res_sp = sp[0];
-    BOOST_CHECK_EQUAL(res_sp.second, to_duration(B.distance_to(S), navitia::type::Mode_e::Walking));
-    BOOST_CHECK_EQUAL(b.data->pt_data->stop_points[res_sp.first]->name, "stop_point:stopB");
+    navitia::routing::map_stop_point_duration tested_map;
+    tested_map.insert(navitia::routing::stop_point_duration(
+                        navitia::routing::SpIdx(*(b.data->pt_data->stop_points_map["stop_point:stopB"])),
+                        to_duration(B.distance_to(S), navitia::type::Mode_e::Walking)));
+    BOOST_CHECK_EQUAL_COLLECTIONS(sp.begin(), sp.end(), tested_map.begin(), tested_map.end());
```
Comment (xlqian) [inline source/routing/tests/routing_api_test.cpp]: if possible: `sp.cbegin()/.cend()` .....

http://stackoverflow.com/questions/12001410/what-is-the-reason-behind-cbegin-cend

and using `std::begin()` is even better:

http://stackoverflow.com/questions/7593086/why-use-non-member-begin-and-end-functions-in-c11

File: source/routing/raptor_api.cpp [hunk 4]
```diff
@@ -748,7 +747,7 @@ get_stop_points( const type::EntryPoint &ep, const type::Data& data,
                     const SpIdx sp_idx = SpIdx(*stop_point);
                     if (stop_points.find(sp_idx) == stop_points.end()) {
                         concerned_path_finder.distance_to_entry_point[sp_idx] = {};
-                        result.push_back({sp_idx, {}});
+                        result.insert(routing::stop_point_duration(sp_idx, {}));
```
Comment (antoine-de) [inline source/routing/raptor_api.cpp]: I think you can remove `stop_points` no ?

File: source/routing/raptor_api.cpp [hunk 5]
```diff
@@ -806,20 +805,22 @@ get_stop_points( const type::EntryPoint &ep, const type::Data& data,
         }
         const auto admin = data.geo_ref->admins[it_admin->second];
 
+        if (! admin->main_stop_areas.empty()) {
+            for (auto stop_area: admin->main_stop_areas) {
+                for(auto sp : stop_area->stop_point_list) {
+                    const SpIdx sp_idx = SpIdx(*sp);
```
Comment (xlqian) [inline source/routing/raptor_api.cpp]: minor advice: `const SpIdx sp_idx{*sp};`   `{}` eliminates the most vexing parser and it tells us that `SpIdx` isn't a function.

**Pull request merged**